sdk/storage/Azure.Storage.Common/src/Shared/AesGcm/AesGcmWindows.cs

```diff
@@ -98,6 +96,14 @@ internal void Decrypt(byte[] nonce, byte[] ciphertext, byte[] tag, byte[] plaint
 
         private static void ThrowIfNotSupported()
         {
+            // Added to prevent using if framework version can use the provided and supported
+            // AES by System.Security.Cryptography library instead
+#if NETSTANDARD2_1_OR_GREATER || NETCOREAPP3_0_OR_GREATER
+            if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                throw Errors.CryptographyAlgorithmNotSupported();
+            }
+#endif
```

Comment:
is this correct ?

Should this always throw on netstandard2.1+ and core 3.0+ ?
And then only throw on non-windows otherwise ?

Would it make sense to compile types in the Shared/AesGcm only if we target netstandard2.0 and carve them out if we compile netstandard2.1+ or core3+? I.e. don't even include these classes in dll that targets platform which has aesgcm built in.
See https://github.com/Azure/azure-sdk-for-net/blob/789a2416e4d7c346f43e51487c0097333c07812a/sdk/core/Azure.Core/src/Pipeline/HttpWebRequestTransport.cs#L20 for example.

Reply:
The intent was to prevent the user from calling into this and use the dotnet runtime AES GCM (or linux preferred version of GCM algorithm).

But you're right, we should just prevent it from compiling in netstandard2.1+ and core3, so they don't have a chance of calling into this code if they have a newer version.

Give me a bit, I'm going to remove this and target the respective platforms.